src/atoms/Switch/Switch.native.js

```diff
@@ -0,0 +1,315 @@
+          },
+        };
+    }
+  },
+  padding({ size, theme }) {
+    switch (size) {
+      case 'medium':
+      default:
+        return theme.spacings.xxsmall;
+      case 'large':
+        return theme.spacings.xsmall;
+    }
+  },
+  alignSelf({ align }) {
+    switch (align) {
+      case 'left':
+      default:
+        return 'flex-start';
+      case 'center':
+        return 'center';
+      case 'right':
+        return 'flex-end';
+    }
+  },
+};
+
+const StyledContainer = styled(TouchableOpacity)(
+  (props) => `
+  height: ${styles.size(props).containerSize.height};
+  width: ${styles.size(props).containerSize.width};
+  border-radius: ${`${parseInt(styles.size(props).containerSize.height, 10) / 2}px`};
+  padding: ${styles.padding(props)};
+  flex-direction: row;
+  align-self: ${styles.alignSelf(props)};
+`,
+);
+
+const StyledKnob = styled(View)(
+  (props) => `
+  position: absolute;
+  align-self: center;
+  height: ${styles.size(props).knobSize.height};
+  border-radius: ${`${parseInt(styles.size(props).knobSize.height, 10) / 2}px`};
+  background-color: #fff;
+`,
+);
+
+const AnimatedContainer = Animated.createAnimatedComponent(StyledContainer);
+const AnimatedKnob = Animated.createAnimatedComponent(StyledKnob);
+
+const moveKnob = (config, cb) => {
+  const [leftSpaceAnim, rightSpaceAnim] = config;
+  Animated.parallel([
+    Animated.timing(leftSpaceAnim.animationProp, {
+      toValue: leftSpaceAnim.toValue,
+      duration: leftSpaceAnim.duration,
+    }),
+    Animated.timing(rightSpaceAnim.animationProp, {
+      toValue: rightSpaceAnim.toValue,
+      duration: rightSpaceAnim.duration,
+    }),
+  ]).start(cb);
+};
+
+const Switch = (props) => {
+  const { disabled, on, defaultOn, onChange, size, align, testID } = props;
+  const theme = useContext(ThemeContext);
+  const {
+    containerSize: { width: containerWidth },
+    knobSize: { width: knobWidth, activeWidth: knobActiveWidth },
+  } = styles.size({ size, theme });
+  const containerPadding = parseInt(styles.padding({ size, theme }), 10);
+  const [toggleState, setToggleState] = useState(false);
+  const [active, setActive] = useState(false);
+
+  const SWITCH_STATE_CONFIGS = {
+    OFF_LEFT_SPACE: containerPadding,
+    OFF_RIGHT_SPACE: parseInt(containerWidth, 10) - containerPadding - parseInt(knobWidth, 10),
+    OFF_ACTIVE_RIGHT_SPACE:
+      parseInt(containerWidth, 10) - containerPadding - parseInt(knobActiveWidth, 10),
+    ON_RIGHT_SPACE: containerPadding,
+    ON_LEFT_SPACE: parseInt(containerWidth, 10) - containerPadding - parseInt(knobWidth, 10),
+    ON_ACTIVE_LEFT_SPACE:
+      parseInt(containerWidth, 10) - containerPadding - parseInt(knobActiveWidth, 10),
+    KNOB_TRANSLATION_DURATION: 200,
+    KNOB_SCALE_DURATION: 100,
+  };
+
+  const leftSpace = useRef(new Animated.Value(SWITCH_STATE_CONFIGS.OFF_LEFT_SPACE));
+  const rightSpace = useRef(new Animated.Value(SWITCH_STATE_CONFIGS.OFF_RIGHT_SPACE));
```

Comment:
Please extract out current for easier understanding of animation. Also rename variable with suffix of ```AnimatedValue``` as a practice so that we know these values are not absolute values but Animated values

```suggestion
  const {current: leftSpaceAnimatedValue} = useRef(new Animated.Value(SWITCH_STATE_CONFIGS.OFF_LEFT_SPACE));
  const {current: rightSpaceAnimatedValue} = useRef(new Animated.Value(SWITCH_STATE_CONFIGS.OFF_RIGHT_SPACE));
```